airbyte-integrations/connectors/destination-cumulio/destination_cumulio/destination.py

```diff
@@ -0,0 +1,118 @@
+
+        :return: Iterable of AirbyteStateMessages wrapped in AirbyteMessage structs.
+        """
+        writer = CumulioWriter(config, configured_catalog, logger)
+
+        for configured_stream in configured_catalog.streams:
+            # Cumul.io does not support removing all data from an existing dataset, and removing the dataset itself will break existing
+            # dashboards built on top of it.
+            # Instead, the connector will make sure to push the first batch of data as a "replace" action: this will cause all existing data
+            # to be replaced with the first batch of data. All next batches will be pushed as an "append" action.
+            if configured_stream.destination_sync_mode == DestinationSyncMode.overwrite:
+                writer.delete_stream_entries(configured_stream.stream.name)
+
+        for message in input_messages:
+            if message.type == Type.STATE:
+                # Yielding a state message indicates that all records which came before it have been written to the destination.
+                # We flush all write buffers in the writer, and then output the state message itself.
+                yield message
+            elif message.type == Type.RECORD:
+                record = message.record
+                assert record is not None
+                assert record.stream is not None
+                assert record.data is not None
+                writer.queue_write_operation(record.stream, record.data)
+            else:
+                # ignore other message types for now
+                continue
+
+        # Make sure to flush any records still in the queue
+        writer.flush_all()
+
+    def check(self, logger: Logger, config: Mapping[str, Any]) -> AirbyteConnectionStatus:
+        """Tests if the input configuration can be used to successfully connect to the destination with the needed permissions.
+
+        This will test whether the combination of the Cumul.io API host, API key and API token is valid.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config: Json object containing the configuration of this destination, content of this json is as specified in
+        the properties of the spec.json file
+
+        :return: AirbyteConnectionStatus indicating a Success or Failure
+        """
+        try:
+            client = CumulioClient(config, logger)
+            # Verify access by hitting Cumul.io authentication endpoint
+            client.test_api_token()
+
+            # We're no longer using testing a data push as this might take some time.
+            # If the API host, key, and token are valid, Data can be pushed
+            # # Verify data push by pushing dummy data into dummy dataset (will be deleted afterwards)
+            # # client.test_data_push(
```

Comment:
please delete unused code and the related declarations above. Feel free to leave the comment in in case someone following up on this connector wonders why it is not trying to write data in `CHECK`